src/gax-internal/src/observability.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::options::InstrumentationClientInfo;
+use gax::options::RequestOptions;
+
+/// Holds information extracted from HTTP requests and responses,
+/// formatted to align with OpenTelemetry semantic conventions for tracing.
+/// This struct is used to populate attributes on tracing spans.
+#[derive(Debug, Clone)]
+#[allow(dead_code)] // TODO(#3239): Remove once used in http.rs
+pub(crate) struct HttpSpanInfo {
+    // Attributes for OpenTelemetry SDK interop
+    otel_kind: String,   // "Client"
+    otel_name: String,   // "{METHOD} {url.template}" or "{METHOD}"
+    otel_status: String, // "Unset", "Ok", "Error"
```

Comment:
This feels like it should be an `enum OtelStatus` or `Option<OtelStatus>` that has a `name() -> &'static str` accessor ?